### PR TITLE
fix set_epoch on TPUs

### PR DESCRIPTION
**Pull request by ibeltagy**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/2622

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/training_loop.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def train(self):
                 if self.reload_dataloaders_every_epoch:
                     self.reset_train_dataloader(model)
                 # set seed for distributed sampler (enables shuffling for each epoch)
-                if (self.use_ddp or self.use_horovod) \
+                if (self.use_ddp or self.use_horovod or self.use_tpu) \
```
Comment (Borda) [inline pytorch_lightning/trainer/training_loop.py:350]: I guess the variable here `on_tpu`

Comment (ibeltagy) [reply pytorch_lightning/trainer/training_loop.py:350]: the `use_tpu` variable is still used everywhere in the codebase. Just curious, when do you use `on_tpu` vs. `use_tpu`?

Comment (Borda) [reply pytorch_lightning/trainer/training_loop.py:350]: I agree we shall clean this up...

**Comment by Borda** [comment]
@ibeltagy mind add test for this case?

**Pull request merged**